--- a/doas.c
+++ b/doas.c
@@ -1,4 +1,11 @@
 #include <sys/types.h>
+#include <limits.h>
+
+#ifndef UID_MAX
+#define UID_MAX 0xffffffff
+#endif
+#ifndef GID_MAX
+#define GID_MAX 0xffffffff
+#endif
 #include <sys/stat.h>
 #include <sys/wait.h>
 #include <sys/ioctl.h>
@@ -8,7 +15,6 @@
 #include <string.h>
 #include <unistd.h>
 #include <pwd.h>
-#include <limits.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <time.h>
