--- openpam-20250531/lib/libpam/openpam_ttyconv.c
+++ openpam-20250531/lib/libpam/openpam_ttyconv.c
@@ -33,6 +33,7 @@
 #include <string.h>
 #include <unistd.h>
 #include <wchar.h>
+#include <stdio_ext.h>
 
 #include <security/pam_appl.h>
 
@@ -290,7 +291,11 @@
 break;
 case PAM_PROMPT_ECHO_ON:
 if (fgets(input, sizeof input, stdin) != NULL) {
+#ifdef __GLIBC__
 fpurge(stdin);
+#else
+__fpurge(stdin);
+#endif
 len = strlen(input);
 if (len > 0 && input[len - 1] == '\n')
 input[--len] = '\0';
