
--- a/compiler/rustc_target/src/spec/linux_musl.rs
+++ b/compiler/rustc_target/src/spec/linux_musl.rs
@@ -6,7 +6,7 @@
 pub fn target(target: &mut Target) {
     target.env = "musl".into();
     target.is_like_musl = true;
-    target.crt_static_default = true;
+    target.static_position_independent_executables = true;
 }
 
 --- a/compiler/rustc_target/src/spec/x86_64_unknown_linux_musl.rs
+++ b/compiler/rustc_target/src/spec/x86_64_unknown_linux_musl.rs
@@ -12,5 +12,5 @@
         "x86_64-unknown-linux-musl",
         Target {
-            crt_static_default: true,
+            crt_static_default: false,
             ..base::opts("linux", "musl")
         },
     )
--- a/compiler/rustc_target/src/spec/aarch64_unknown_linux_musl.rs
+++ b/compiler/rustc_target/src/spec/aarch64_unknown_linux_musl.rs
@@ -12,7 +12,7 @@
         "aarch64-unknown-linux-musl",
         Target {
-            crt_static_default: true,
+            crt_static_default: false,
             ..base::opts("linux", "musl")
         },
     )
--- a/compiler/rustc_target/src/spec/riscv64gc_unknown_linux_musl.rs
+++ b/compiler/rustc_target/src/spec/riscv64gc_unknown_linux_musl.rs
@@ -12,7 +12,7 @@
         "riscv64gc-unknown-linux-musl",
         Target {
-            crt_static_default: true,
+            crt_static_default: false,
             ..base::opts("linux", "musl")
         },
     )
--- a/compiler/rustc_target/src/spec/mod.rs
+++ b/compiler/rustc_target/src/spec/mod.rs
@@ -1656,6 +1656,12 @@
     ("x86_64-unknown-linux-musl", x86_64_unknown_linux_musl::target),
     ("x86_64-unknown-netbsd", x86_64_unknown_netbsd::target),
     ("x86_64-unknown-none", x86_64_unknown_none::target),
+    ("x86_64-museix-linux-musl", x86_64_museix_linux_musl::target),
+    ("aarch64-museix-linux-musl", aarch64_museix_linux_musl::target),
+    ("riscv64gc-museix-linux-musl", riscv64gc_museix_linux_musl::target),
+    ("armv7-museix-linux-musleabihf", armv7_museix_linux_musleabihf::target),
+    ("armv7-museix-linux-musleabi", armv7_museix_linux_musleabi::target),
+    ("arm-museix-linux-musleabi", arm_museix_linux_musleabi::target),
     ("x86_64-unknown-nto-qnx710", x86_64_unknown_nto_qnx710::target),
     ("x86_64-unknown-openbsd", x86_64_unknown_openbsd::target),
     ("x86_64-unknown-redox", x86_64_unknown_redox::target),
@@ -2022,6 +2028,12 @@
 pub mod x86_64_unknown_linux_musl;
 pub mod x86_64_unknown_netbsd;
 pub mod x86_64_unknown_none;
+pub mod x86_64_museix_linux_musl;
+pub mod aarch64_museix_linux_musl;
+pub mod riscv64gc_museix_linux_musl;
+pub mod armv7_museix_linux_musleabihf;
+pub mod armv7_museix_linux_musleabi;
+pub mod arm_museix_linux_musleabi;
 pub mod x86_64_unknown_nto_qnx710;
 pub mod x86_64_unknown_openbsd;
 pub mod x86_64_unknown_redox;
